be/src/common/s3_uri.cpp

```diff
@@ -37,10 +39,17 @@ bool S3URI::parse(const char* uri_str) {
         path = uri.path();
     }
 
-    if (host.find('.') == std::string::npos) {
+    if (_scheme == "s3" || _scheme == "s3a" || _scheme == "s3n") {
+        // s3 style format: https://docs.aws.amazon.com/solutions/latest/media2cloud-on-aws/file-paths-in-amazon-s3.html
         // URL like S3://bucket-name/key-name
         _bucket = host;
         _key = path;
+    } else if (host.find('.') == std::string::npos) {
+        // TODO We need to check each cloud vendor's path style, like ks3, tos, ..., etc
+        // OSS's path format: https://help.aliyun.com/document_detail/154985.html and https://help.aliyun.com/document_detail/415351.html
```

Comment:
does OSS allow `.` in bucket name?

Reply:
It's not allowed, but it has its own rules, it's really different.